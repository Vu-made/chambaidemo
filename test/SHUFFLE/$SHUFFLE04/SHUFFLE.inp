100 10
69 16
2 67
95 22
28 85
18 69
89 69
23 17
32 45
82 87
7 30
