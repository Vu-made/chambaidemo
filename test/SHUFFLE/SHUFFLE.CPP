#include <bits/stdc++.h>
#define N 100005
using namespace std;
int Next[N], Pre[N];
int n,x;
int main()
{
    freopen("SHUFFLE.INP","r",stdin);
    freopen("SHUFFLE.OUT","w",stdout);
    cin >> n >> x;
    Next[0] = 1; Pre[n+1] = n;
    for (int i=1;i<=n; i++)
    {
        Next[i] = i+1;
        Pre[i] = i-1;
    }
    for (int i,j,k=1;k <= x;k++)
    {
        cin >> i >> j;
        Next[Pre[i]] = Next[i];
        Pre[Next[i]] = Pre[i];
        Next[i] = j;
        Pre[i] = Pre[j];
        Next[Pre[j]] = i;
        Pre[j] = i;
    }
    for (int p=Next[0]; p<n+1; p=Next[p])
        cout << p << " ";
    return 0;
}
