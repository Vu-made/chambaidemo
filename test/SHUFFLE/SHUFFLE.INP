9 3
8 2
4 7
1 9
