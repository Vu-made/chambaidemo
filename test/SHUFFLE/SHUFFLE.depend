# depslib dependency file v1.0
1602671316 source:c:\users\admin\desktop\bgdv\bt\thuat_toan\array1d_3\shuffle\shuffle.cpp
	<bits/stdc++.h>

