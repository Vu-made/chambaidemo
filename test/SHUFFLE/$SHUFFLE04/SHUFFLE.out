1 3 4 5 6 8 9 10 11 12 13 14 15 18 89 69 16 23 17 19 20 21 95 22 24 25 26 27 29 7 30 31 33 34 35 36 37 38 39 40 41 42 43 44 32 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 2 67 68 70 71 72 73 74 75 76 77 78 79 80 81 83 84 28 85 86 82 87 88 90 91 92 93 94 96 97 98 99 100 